File: order.service/src/main/java/br/com/edu/order/infra/rest/v1/response/ListResponse.java
/* (C)2024 */
package br.com.edu.order.infra.rest.v1.response;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.util.List;
import lombok.Builder;
import lombok.Data;

@Builder
@Data
public class ListResponse {

    @JsonProperty("content")
    private List<OrderResponse> content;

    @JsonProperty("page")
    private Page page;
}

File: order.service/order-service-app/src/main/kotlin/br/com/edu/order/infra/rest/v1/response/ListResponse.kt
package br.com.edu.order.infra.rest.v1.response

import com.fasterxml.jackson.annotation.JsonInclude
import com.fasterxml.jackson.annotation.JsonProperty

@JsonInclude(JsonInclude.Include.NON_NULL)
data class ListResponse(

    @JsonProperty("content")
    val content: List<OrderResponse> = listOf(),

    @JsonProperty("page")
    val page: Page = Page()
)